### ISSUE TITLE
Loss is zero

### ISSUE THREAD

**Issue by bernardus-brouwer** [LMissher/TFMAE#12]
Hi!

I have been trying to reproduce your results, but unfortunately I am not able to because the loss function is zero.

`loss =  con_loss - adv_loss` is always zero because `con_loss` cancels out `adv_loss`. I encounter this issue on all datasets.

As far as I understand it, this is expected behavior, given the code as it is:

> `
>                     for u in range(len(freatt)):
>                         adv_loss += (torch.mean(my_kl_loss(tematt[u], (
>                                 freatt[u] / torch.unsqueeze(torch.sum(freatt[u], dim=-1), dim=-1)).detach())) + torch.mean(
>                             my_kl_loss((freatt[u] / torch.unsqueeze(torch.sum(freatt[u], dim=-1), dim=-1)).detach(),
>                                     tematt[u])))
>                         con_loss += (torch.mean(my_kl_loss(
>                             (freatt[u] / torch.unsqueeze(torch.sum(freatt[u], dim=-1), dim=-1)),
>                             tematt[u].detach())) + torch.mean(
>                             my_kl_loss(tematt[u].detach(), (
>                                     freatt[u] / torch.unsqueeze(torch.sum(freatt[u], dim=-1), dim=-1)))))
> 
>                     adv_loss = adv_loss / len(freatt)
>                     con_loss = con_loss / len(freatt)
> 
>                     loss =  con_loss - adv_loss
> `

Even for arbitrary `tematt` and `freatt`, the loss will always be zero.

Any suggestions?

**Comment by LMissher** [comment]
Please refer previous issue #7
